### PR TITLE
AWS DBs Heartbeat: return IAM status

### PR DESCRIPTION
**Pull request by marcoandredinis**
## Why
During the Discover flow for adding AWS Databases, we ask the user to set up the IAM Policy.
However, sometimes that's not required:
- they set it up previously
- database service has access to AWS APIs to create the policy by itself

For those cases, the UI must skip that step.

## How
When heartbeating the DatabaseServer back, it will set the `database.aws.iam_policy_exists` field to:
- true if the the IAM Policy was created or already exists
- false if there was an error when it tried to create the Policy

Demo:
When the policy was auto-created by the DatabaseService.
image

After manually removing the policy and removing policies to deny IAM Management to the DatabaseService:
image

### CODE REVIEW

File: tool/tsh/common/tsh_test.go [hunk 1]
```diff
@@ -3524,6 +3524,7 @@ func TestSerializeDatabases(t *testing.T) {
         "rds": {
           "iam_auth": false
         },
+		"iam_policy_exists": false,
```
Comment (strideynet) [inline tool/tsh/common/tsh_test.go:3527]: This new entry seems weirdly indented 😓

**Review by strideynet — APPROVED** [review]
Looks good - just some weird indentation in `tsh_test.go`

File: tool/tsh/common/tsh_test.go (hunk 1 above)
Comment (marcoandredinis) [reply tool/tsh/common/tsh_test.go:3527]: Thank you, I'll fix it.
My editor failed me, I'm disappointed.

**Comment by marcoandredinis** [comment]
@EdwardDowling can you please take a look when you get a chance?

**Review by EdwardDowling — APPROVED** [review]

**Pull request merged**